cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1458,18 +1463,17 @@ fromLedgerTxInsCollateral
   -> TxInsCollateral era
 fromLedgerTxInsCollateral era body =
     case collateralSupportedInEra (shelleyBasedToCardanoEra era) of
-      Nothing        -> TxInsCollateralNone
+      Nothing             -> TxInsCollateralNone
+      _ | null collateral -> TxInsCollateralNone
       Just supported -> TxInsCollateral supported
                           [ fromShelleyTxIn input
-                          | input <- Set.toList (collateral era body) ]
+                          | input <- Set.toList collateral ]
   where
-    collateral :: ShelleyBasedEra era
-               -> Ledger.TxBody (ShelleyLedgerEra era)
-               -> Set (Shelley.TxIn StandardCrypto)
-    collateral ShelleyBasedEraShelley = const Set.empty
-    collateral ShelleyBasedEraAllegra = const Set.empty
-    collateral ShelleyBasedEraMary    = const Set.empty
-    collateral ShelleyBasedEraAlonzo  = Alonzo.collateral'
+    collateral = case era of
+      ShelleyBasedEraShelley -> Set.empty
+      ShelleyBasedEraAllegra -> Set.empty
+      ShelleyBasedEraMary    -> Set.empty
+      ShelleyBasedEraAlonzo  -> Alonzo.collateral' body
```

Comment:
Any reason for this change?

Reply:
While I was reading this code, I reformatted it to help myself understand it. I think it's simpler and clear now. Do you think it has some performance downsides?

Reply:
I think we should leave a type signature